src/Symfony/Component/Messenger/Worker.php

```diff
@@ -99,6 +102,10 @@ public function run(array $options = []): void
                 if ($queueNames) {
                     $envelopes = $receiver->getFromQueues($queueNames);
                 } else {
+                    if (method_exists($receiver, 'getMessageGenerator')) {
```

Comment:
I don't think we'll want the messenger to be aware of the scheduler. Ideally, we want these events dispatched in the scheduler but can't currently think of how.